core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -32,48 +33,64 @@ import scala.jdk.CollectionConverters._
 
 class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSugar with TestDriverComponent {
 
-  "SnapshotService" should {
-    "create a new snapshot reference to a TDR snapshot" in withMinimalTestDatabase { _ =>
-      val mockSamDAO = mock[SamDAO](RETURNS_SMART_NULLS)
-      when(
-        mockSamDAO.userHasAction(ArgumentMatchers.eq(SamResourceTypeNames.workspace),
-                                 any[String],
-                                 any[SamResourceAction],
-                                 any[RawlsRequestContext]
-        )
-      ).thenReturn(Future.successful(true))
-      when(
-        mockSamDAO.getUserStatus(any[RawlsRequestContext])
-      ).thenReturn(
-        Future.successful(
-          Some(SamUserStatusResponse(userInfo.userSubjectId.value, userInfo.userEmail.value, enabled = true))
-        )
+  // create a mockito-powered SamDAO that always returns true for permission checks and returns
+  // a test-fixture user info object
+  private def defaultMockSamDao() = {
+    val mockSamDAO = mock[SamDAO](RETURNS_SMART_NULLS)
+
+    when(
+      mockSamDAO.userHasAction(ArgumentMatchers.eq(SamResourceTypeNames.workspace),
+                               any[String],
+                               any[SamResourceAction],
+                               any[RawlsRequestContext]
+      )
+    ).thenReturn(Future.successful(true))
+    when(
+      mockSamDAO.getUserStatus(any[RawlsRequestContext])
+    ).thenReturn(
+      Future.successful(
+        Some(SamUserStatusResponse(userInfo.userSubjectId.value, userInfo.userEmail.value, enabled = true))
       )
+    )
 
-      val mockWorkspaceManagerDAO = mock[WorkspaceManagerDAO](RETURNS_SMART_NULLS)
-      when(
-        mockWorkspaceManagerDAO.createDataRepoSnapshotReference(
-          any[UUID],
-          any[UUID],
-          any[DataReferenceName],
-          any[Option[DataReferenceDescriptionField]],
-          any[String],
-          any[CloningInstructionsEnum],
-          any[RawlsRequestContext]
-        )
+    mockSamDAO
+  }
+
+  // create a mockito-powered WorkspaceManagerDAO that always returns success for creating
+  // a snapshot reference. Tests can add more responses/expectations to the mock returned by this method.
+  private def defaultMockWorkspaceManagerDao() = {
```

Comment:
maybe to support my comment later on, you can pass in a `workspaceId` from the caller, and use that instead of something randomly generated -- then you can look for a hardcoded string in later verification